### ISSUE TITLE
502 Bad Gateway after cPanel/WHM updated to v90

### ISSUE THREAD

**Issue by stratacorps** [engintron/engintron#1221]
I found similar threads but since engintron had been working fine until this morning. Today WHM was updated to 90.0.2 and all sites were down with 502 Bad Gateway.

In the logs we have:

SSL_do_handshake() failed (SSL: error:1407742E:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert protocol version) while SSL handshaking to upstream

We've tried setting the IP address in proxy_pass in the proxy_params_common with no results. The only solution for now is to disable engintron.

**Comment by fevangelou** [comment]
See this: https://serverfault.com/questions/720745/ssl-handshake-failed-with-nginx

If you've modified Engintron's config files, re-install it to reset its settings.

**Comment by stratacorps** [comment]
I removed engintron (engintron.sh remove), re-installed it, and the same. From the thread above I looked and cPanel has TLS v1.2 enabled as default--we had previously disabled older v1/1.1. Not sure why the cpanel update would make apache stop responding to Nginx, but I changed things back to allow for intermediate client support 1.2/1.3 and copied the ssl_ciphers in the nginx config to apache. Apache accepts everything except v2/v3/v1/v1.1. I understand the error better now...nginx talks to apache and if nginx passes a protocol over to apache that apache doesn't accept, it gets blocked.  Why this had worked before the update of WHM I don't know.
